[FSDP Optim State] Remove checkpoint prefix

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #80480

Remove `_checkpoint_wrapped_module` prefixes when creating keys for optimizer state_dict.

Having these does not actually create an issue for optim_state_dict save / load, but we'd like to strip these keys out for downstream code that consumes these APIs typically expecting checkpointing prefixes to not exist (as checkpointing should be a transparent operation which should not change module / parameter names).

rohan-varma committed:
Remove checkpoint prefix

[ghstack-poisoned]

awgu commented on test/distributed/fsdp/test_fsdp_optim_state.py:
```diff
@@ -477,12 +480,21 @@ def test_full_optim_state_dict_keys(self):
         self.assertEqual(optim_state_dict["state"].keys(), state_dict.keys())
+        # Check that checkpointing prefix was indeed stripped.
+        for key in optim_state_dict["state"].keys():
+            self.assertFalse(_CHECKPOINT_PREFIX in key)
```
nit: 😄 
```suggestion
            self.assertNotIn(_CHECKPOINT_PREFIX, key)
```
https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertNotIn

awgu reviewed (approved):
Thanks for the super fast fix. This looks good to me!

awgu commented on test/distributed/fsdp/test_fsdp_optim_state.py:
```diff
@@ -477,12 +480,21 @@ def test_full_optim_state_dict_keys(self):
             state_dict = wrapped_model.state_dict()
         self.assertEqual(optim_state_dict["state"].keys(), state_dict.keys())
+        # Check that checkpointing prefix was indeed stripped.
+        for key in optim_state_dict["state"].keys():
```
one more nit:
```suggestion
        for key in optim_state_dict["state"]:
```

rohan-varma committed:
Update on "[FSDP Optim State] Remove checkpoint prefix"


Remove `_checkpoint_wrapped_module` prefixes when creating keys for optimizer state_dict.

Having these does not actually create an issue for optim_state_dict save / load, but we'd like to strip these keys out for downstream code that consumes these APIs typically expecting checkpointing prefixes to not exist (as checkpointing should be a transparent operation which should not change module / parameter names).

[ghstack-poisoned]

fegin reviewed (approved):
LGTM

rohan-varma commented:
@pytorchbot merge

rohan-varma committed:
Update on "[FSDP Optim State] Remove checkpoint prefix"


Remove `_checkpoint_wrapped_module` prefixes when creating keys for optimizer state_dict.

Having these does not actually create an issue for optim_state_dict save / load, but we'd like to strip these keys out for downstream code that consumes these APIs typically expecting checkpointing prefixes to not exist (as checkpointing should be a transparent operation which should not change module / parameter names).

[ghstack-poisoned]

rohan-varma commented:
@pytorchbot merge

rohan-varma committed:
Update on "[FSDP Optim State] Remove checkpoint prefix"


Remove `_checkpoint_wrapped_module` prefixes when creating keys for optimizer state_dict.

Having these does not actually create an issue for optim_state_dict save / load, but we'd like to strip these keys out for downstream code that consumes these APIs typically expecting checkpointing prefixes to not exist (as checkpointing should be a transparent operation which should not change module / parameter names).

[ghstack-poisoned]

rohan-varma commented:
CI failures are unrelated

rohan-varma commented:
@pytorchbot merge -f

suo commented:
@pytorchbot merge -m "Broke master https://hud.pytorch.org/pytorch/pytorch/commit/04c50fec1ceec5b2d78934921f1cb366ff3fac81, the test failures were not unrelated" -c ignoredsignal

suo commented:
@pytorchbot revert -m "Broke master https://hud.pytorch.org/pytorch/pytorch/commit/04c50fec1ceec5b2d78934921f1cb366ff3fac81, the test failures were not unrelated" -c ignoredsignal

rohan-varma commented:
@suo my bad I guess the PR is somehow causing the issue in https://hud.pytorch.org/pytorch/pytorch/commit/04c50fec1ceec5b2d78934921f1cb366ff3fac81, it is very strange since none of those failing APIs are touched in this PR.
